app/item/controller.py

```diff
@@ -1,4 +1,104 @@
 from flask import Blueprint, request, g, redirect, url_for, jsonify
 from datetime import datetime, timedelta
+import pprint
 
-from app.auth.decorators import requires_login, requres_status_manager
+from app import db
+from app.item.models.Item import Item
+from app.auth.models.User import User
+from app.item.validators.ItemValidator import ItemValidator
+#from app.auth.decorators import requires_login, requres_status_manager
+
+
+
+itemMod = Blueprint('item', __name__, url_prefix='/item')
+
+##index
+@itemMod.route('/', methods=['GET', 'POST'])
+def index():
+    #index page
+    if request.method == 'GET':
+        return jsonify({
+            'legal routes': {
+                'route(\'/\')': 'for methods=[\'GET\',\'POST\']',
+                'route(\'/<string:itemName>\')': 'for methods=[\'GET\', \'PUT\', \'DELETE\']'
+                }
+        }), 201
+
+    # add item to database
+    if request.method == 'POST':
+        print("got into add_item()")
+        # validate the inputted information.
+        form = ItemValidator(data=request.json)
+        if form.validate():
+            # Check whether item exists
+            name = request.json['name']
+            check_item = get(name)
+            if check_item:
+                return jsonify(
+                    '%r already exists in your system.' % name,{
+                    'item': a_dict(check_item)
+                    }), 400
+            #add item to database
+            item = Item(request.json['name'], request.json['cost'], request.json['description'])
+            db.session.add(item)
+            db.session.commit()
+            return jsonify(
+                'Item added successfully.',
+                { 'added item': item.to_dict()
+            }), 201
+        return jsonify({'message': 'There was an error adding data', 'error': form.errors}), 400
+
+@itemMod.route('/<string:itemName>', methods=['GET', 'PUT', 'DELETE'])
```

Comment:
we should use item id to avoid any kinds of sanitation problem so maybe

```python
@itemMod.route('/<int:itemId>', methods=['GET','PUT','DELETE'])
```